homeassistant/components/zwave_js/humidifier.py

```diff
@@ -0,0 +1,225 @@
+
+        return val
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Set new target humidity."""
+        if self._current_mode:
+            if (
+                str(HumidityControlMode.AUTO.value)
+                in self._current_mode.metadata.states
+            ):
+                new_mode = HumidityControlMode.AUTO
+            elif (
+                str(HumidityControlMode.HUMIDIFY.value)
+                in self._current_mode.metadata.states
+            ):
+                new_mode = HumidityControlMode.HUMIDIFY
+            else:
+                new_mode = HumidityControlMode.DEHUMIDIFY
+            await self.info.node.async_set_value(self._current_mode, new_mode)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Set new target humidity."""
+        if self._current_mode:
+            await self.info.node.async_set_value(
+                self._current_mode, HumidityControlMode.OFF
+            )
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return True if entity is on."""
+        if self._current_mode is None:
+            return None
+        return cast(bool, int(self._current_mode.value) != HumidityControlMode.OFF)
+
+    @property
+    def target_humidity(self) -> int | None:
+        """Return the humidity we try to reach."""
+        if not self.is_on or self._setpoint_value is None:
+            return None
+        return int(self._setpoint_value.value)
+
+    @property
+    def mode(self) -> str | None:
+        """Return the current humidity control mode."""
+        if self._current_mode:
+            return cast(
+                str, self._current_mode.metadata.states[str(self._current_mode.value)]
+            )
+        return None
+
+    async def async_set_mode(self, mode: str) -> None:
+        """Set new mode."""
+
+        if self._current_mode is None:
+            return None
+
+        try:
+            new_state = int(
+                next(
+                    state
+                    for state, label in self._current_mode.metadata.states.items()
+                    if label == mode
+                )
+            )
+        except StopIteration:
+            raise ValueError(f"Received an invalid mode: {mode}") from None
+
+        await self.info.node.async_set_value(self._current_mode, new_state)
+
+    async def async_set_humidity(self, humidity: int) -> None:
+        """Set new target humidity."""
+        await self.info.node.async_set_value(self._setpoint_value, humidity)
+
+    @property
+    def min_humidity(self) -> int:
+        """Return the minimum humidity."""
+        min_value = DEFAULT_MIN_HUMIDITY
+        try:
+            if self._setpoint_value and self._setpoint_value.metadata.min:
+                min_value = self._setpoint_value.metadata.min
+        # In case of any error, we fallback to the default
+        except (IndexError, ValueError, TypeError):
```

Comment:
When can these exceptions be raised?

Reply:
This was copied from `climate.py` but I think this was added because of the temperature conversion, it may not be necessary here. I removed it.